src/v/container/interval_set.h

```diff
@@ -34,12 +34,15 @@ class interval_set {
     using set_t = absl::btree_map<T, T>;
 
 public:
+    using key_type = set_t::key_type;
+    using mapped_type = set_t::mapped_type;
```

Comment:
key and mapped type are always the same and should be replaced with just T I think